projects/admin/src/app/shared/preview-email/component/preview-email/preview-email.component.ts

```diff
@@ -0,0 +1,132 @@
+  @Input() previewPosition: 'top' | 'bottom' = 'top';
+  /** Available types of destination */
+  @Input() emailTypes: ('to' | 'cc' | 'bcc' | 'reply_to')[] = ['to', 'cc', 'bcc', 'reply_to'];
+  /** Mandatory types */
+  @Input() mandatoryEmailTypes: ('to' | 'cc' | 'bcc' | 'reply_to')[] = ['to'];
+  /** Allows to pre-populate the recipients */
+  @Input() set prePopulateRecipients(rep: ITypeEmail[]) {
+    const keys = Object.keys(this.recipients);
+    rep.forEach((el: ITypeEmail) => {
+      if (keys.includes(el.type)) {
+        this.recipients[el.type].push(el.address);
+      }
+    });
+  }
+  /** Event to transmit email address recipients */
+  @Output() data = new EventEmitter<ITypeEmail[]>();
+  /** Event to allow closing the dialog */
+  @Output() closeDialog = new EventEmitter<boolean>(false);
+
+  /** Email address that is in drag mode */
+  draggedEmail: string;
+
+  /** Recipients */
+  recipients = { to: [], cc: [], bcc: [], reply_to: [] };
+
+  /**
+   * Drag start
+   * @param email - string
+   */
+  dragStart(email: string): void {
+    this.draggedEmail = email;
+  }
+
+  /** Drag end */
+  dragEnd(): void {
+    this.draggedEmail = null;
+  }
+
+  /**
+   * Adding the email address to a specific recipient.
+   * @param field - DragEvent
+   */
+  drop(field: DragEvent): void {
+    if (this.draggedEmail) {
+      const fieldId = field.target['id'];
+      if (fieldId && !this.recipients[fieldId].some((email: string) => email === this.draggedEmail)) {
+        this.recipients[fieldId].push(this.draggedEmail);
+      }
+    }
+  }
+
+  /** Closing the dialog */
+  close(): void {
+    this.closeDialog.emit(true);
+  }
+
+  /**
+   * Confirmation of dialogue data.
+   * Transformation of simple email addresses into TypeEmail type and data transmission.
+   * The closing of the dialog takes place in the parent if the transmitted data are valid.
+   */
+  confirm(): void {
+    const recipients = [];
+    Object.keys(this.recipients).forEach((key: string) => this.recipients[key].every((email: string) => recipients.push({ type: key, address: email })));
+    this.data.emit(recipients);
+  }
+
+  /**
+   * Data validation.
+   * @returns true if the data of the containers are valid.
+   */
+  formValid(): boolean {
+    // Validation of the mandatory types
+    if (!this.mandatoryEmailTypes.every((key: string) => this.recipients[key].length > 0)) {
+      return false;
+    }
+    // Validation of all email addresses of each recipient
+    return Object.keys(this.recipients).every((key: string) => this.recipients[key].every(email => Tools.validateEmail(email)));
+  }
+}
+
+export interface ITypeEmail {
+  type: string;
+  address: string;
+}
+
+export interface IRecipientSuggestion {
+  address: string;
+  type?: string[];
+}
+
+export interface ISuggestions {
+  emails: string[];
+  recipients: ITypeEmail[];
+}
+
+export interface IPreview {
+  preview: string;
+  recipient_suggestions: IRecipientSuggestion[];
+}
```

Comment:
In my opinion, I should be of interest to outsource interfaces into a specific file